feat: Multi list indentation at once

Ahad-patel opened the pull request:
Closes : https://github.com/AppFlowy-IO/AppFlowy/issues/6255

This PR solves the issue of mulitple indentation of array lists at once. 

https://github.com/user-attachments/assets/223c1378-8cd4-4eb1-ac1a-62290232d9ad

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AppFlowy-IO/appflowy-editor?pullRequest=903) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/AppFlowy-IO/appflowy-editor?pullRequest=903) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AppFlowy-IO/appflowy-editor?pullRequest=903) it.</sub>

Ahad-patel commented:
Hey, @Xazin @LucasXu0  

I have added the feature for mulit list indentation at once, some of the tests are failing can you help me solve these issues?

gouravbambhore commented on lib/src/editor/block_component/base_component/indent_command.dart:
Good contribution !

Ahad-patel replied:
Thanks @gouravbambhore,

but some of the checks are failing can you help me resolve it?

LucasXu0 commented on lib/src/editor/block_component/base_component/indent_command.dart:
```diff
@@ -23,41 +23,64 @@ final CommandShortcutEvent indentCommand = CommandShortcutEvent(
 
 bool isIndentable(EditorState editorState) {
   final selection = editorState.selection;
-  if (selection == null || !selection.isSingle) {
```
This condition `!selection.isSingle` will prevent the single line from being indented.

LucasXu0 commented:
@Ahad-patel There's one more case that should be considered.

The `getNodesInSelection` function will return all the nodes, including the children. For example, the tab command doesn't work in the case shown in the attached screenshot. When judging if the nodes are indentable, we just need to consider the first level and filter out their children nodes.

And please add more tests to cover this case.

<img width="637" alt="Screenshot 2024-09-30 at 10 23 08" src="https://github.com/user-attachments/assets/0b05606f-8e57-48b7-a267-9d6e63b09414">

Ahad-patel commented:
Hey @LucasXu0, 

I have updated outdent_command file according to your changes, I have also added tests cases for indent_command and outdent_command.

 I noticed that the outdent_command test file was named "outdent_handler_test.dart" ,since other test files were named according to their file name, so I renamed it to "outdent_command_test.dart".   

I also removed isAllOnSameLevel because we are ignoring child nodes, so the isAllOnSameLevel method will always be true.

Are there any more remaining tasks in this PR?

LucasXu0 commented:
> I noticed that the outdent_command test file was named "outdent_handler_test.dart" ,since other test files were named according to their file name, so I renamed it to "outdent_command_test.dart".

Good catch.

> Are there any more remaining tasks in this PR?

I'm checking the newly added code and will comment on the PR.

LucasXu0 commented on test/new/service/shortcuts/command_shortcut_events/outdent_command_test.dart:
```diff
@@ -212,4 +212,167 @@ void main() async {
+        (tester) async {
+      const text = 'Welcome to Appflowy 😁';
+      final editor = tester.editor
+        // final editor = tester.editor..addNode(paraNode);
```
remove this line.

Ahad-patel replied:
done

LucasXu0 commented on test/new/service/shortcuts/command_shortcut_events/outdent_command_test.dart:
```diff
@@ -212,4 +212,167 @@ void main() async {
+        (tester) async {
+      const text = 'Welcome to Appflowy 😁';
+      final editor = tester.editor
+        // final editor = tester.editor..addNode(paraNode);
```
remove this line

Ahad-patel replied:
done

LucasXu0 commented:
> There is a selection before and after the line when this is in selection editorState.getNodesInSelection will give both the nodes.

It works as intended. You can see the toolbar location. Actually, the previous line has been selected as well. An improvement we can make here is to add a visual selection area for the previous line.

<img width="253" alt="Screenshot 2024-10-04 at 11 10 12" src="https://github.com/user-attachments/assets/907c1b00-09d9-4a9d-9f33-608c83dfc118">

Ahad-patel commented:
> An improvement we can make here is to add a visual selection area for the previous line.

Should I make an new Issue and new PR for this ?

Xazin commented:
> > There is a selection before and after the line when this is in selection editorState.getNodesInSelection will give both the nodes.
> 
> It works as intended. You can see the toolbar location. Actually, the previous line has been selected as well. An improvement we can make here is to add a visual selection area for the previous line.
> 


Normally if you have the cursor before the last character, and you use `Shift + arrow right`, it'll select the last character without the newline (so not the next line). This should be seen as a bug.

Ahad-patel commented:
> Normally if you have the cursor before the last character, and you use `Shift + arrow right`, it'll select the last character without the newline (so not the next line). This should be seen as a bug.

I have created a new issue for this https://github.com/AppFlowy-IO/AppFlowy/issues/6500
